core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -403,32 +403,47 @@ class DAGScheduler(
     parents.toList
   }
 
-  /** Find ancestor shuffle dependencies that are not registered in shuffleToMapStage yet */
-  private def getAncestorShuffleDependencies(rdd: RDD[_]): Stack[ShuffleDependency[_, _, _]] = {
-    val parents = new Stack[ShuffleDependency[_, _, _]]
+  /**
+   * Find ancestor shuffle dependencies that are not registered in shuffleToMapStage yet.
+   * This is done in topological order to create ancestor stages first to ensure that the result
```

Comment:
Can you move this comment to some place inside the method? It doesn't seem relevant to someone using this method, since the order of the ShuffleDependencies returned doesn't matter